targets/esp8266/source/jerry_run.c

```diff
@@ -28,93 +28,96 @@ static const char* fn_sys_loop_name = "sysloop";
 
 int js_entry (const char *source_p, const size_t source_size)
 {
-  const jerry_api_char_t *jerry_src = (const jerry_api_char_t *) source_p;
-  jerry_completion_code_t ret_code = JERRY_COMPLETION_CODE_OK;
-  jerry_flag_t flags = JERRY_FLAG_EMPTY;
-  jerry_api_object_t *err_obj_p = NULL;
+  const jerry_char_t *jerry_src = (const jerry_char_t *) source_p;
+  int ret_code = 0; /* JERRY_COMPLETION_CODE_OK */
+  jerry_init_flag_t flags = JERRY_INIT_EMPTY;
+  jerry_value_t parsed_code;
+
 
   jerry_init (flags);
 
   js_register_functions ();
 
-  if (!jerry_parse ((jerry_api_char_t *)jerry_src, source_size, &err_obj_p))
+  parsed_code = jerry_parse ((jerry_char_t *)jerry_src, source_size,false);
+
+  if (jerry_value_has_error_flag (parsed_code))
   {
     printf ("Error: jerry_parse failed\r\n");
-    ret_code = JERRY_COMPLETION_CODE_UNHANDLED_EXCEPTION;
-  }
-  else
-  {
-    if ((flags & JERRY_FLAG_PARSE_ONLY) == 0)
-    {
-      jerry_api_value_t err_value = jerry_api_create_void_value ();
-      ret_code = jerry_run (&err_value);
-    }
+    ret_code = JERRY_ERROR_SYNTAX;
   }
+  jerry_release_value(parsed_code);
 
   return ret_code;
 }
 
 int js_eval (const char *source_p, const size_t source_size)
 {
-  jerry_completion_code_t status;
-  jerry_api_value_t res;
+  int status = 0;
+  jerry_value_t parsed_code;
 
-  status = jerry_api_eval ((jerry_api_char_t *) source_p,
+  parsed_code = jerry_eval ((jerry_char_t *) source_p,
```

Comment:
parsed_code is a bit misleading variable name. The return value of eval is not a bytecode, but the result of the given source, because eval parse and execute the given JS source code.